linked_list.py

```diff
@@ -17,13 +17,16 @@ def insert(self, val):
     def pop(self):
         # make a node that has the value of the head
         # make the head's head the head
-        if self.head is not None:
+        # if self.head is not None:
+        try:
             head_val = self.head.val
             self.head = self.head.next
             return head_val
+        except AttributeError:
+            return "No head"
```

Comment:
Better to raise an error than to return a string.  How would this be used in code?

```python
val = linked_list.pop()
if val == 'No head':
    val = "default"
```

Wouldn't it be better to say:

```python
try:
    val = linked_list.pop()
except ValueError:
    val = 'default'
```